### PR TITLE
cmake: use GNUInstallDirs

### PR DESCRIPTION
**Pull request by vmatare**
Remove hardcoded installation paths and use GNUInstallDirs, which should provide sane & standards-conformant values.

### CODE REVIEW

**Comment by vmatare** [comment]
Just noticed that this will conflict with PR #447. Ping me if/when I should do a rebase.

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,10 +70,10 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${g2o_LIBRARY_OUTPUT_DIRECTORY})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${g2o_RUNTIME_OUTPUT_DIRECTORY})
 
 # Set standard installation directories
-set(RUNTIME_DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
-set(LIBRARY_DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
-set(ARCHIVE_DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
-set(INCLUDES_DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+set(RUNTIME_DESTINATION ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR})
```
Comment (RainerKuemmerle) [inline CMakeLists.txt:73]: CMAKE_INSTALL_FULL_BINDIR instead seems like a better fit to also handle some special cases.

**Review by RainerKuemmerle — CHANGES_REQUESTED** [review]
Thanks for providing the PR. The FULL variables from CMake seem to be a better fit.

File: CMakeLists.txt (hunk 1 above)
Comment (RainerKuemmerle) [reply CMakeLists.txt:73]: Or as #447 suggest, we should only use relative?

Comment (siposcsaba89) [reply CMakeLists.txt:73]: Hi,

CMake recommends to use relative paths:

"As absolute paths are not supported by cpack installer generators, it is preferable to use relative paths throughout. In particular, there is no need to make paths absolute by prepending CMAKE_INSTALL_PREFIX; this prefix is used by default if the DESTINATION is a relative path."

Best,
Csaba

Comment (vmatare) [reply CMakeLists.txt:73]: I'm totally in support of not overspecifying things, so I guess I'll just rebase onto #447 and make everything relative.

**Comment by vmatare** [comment]
OK, now I'm not entirely sure what to do about all these `INSTALL_NAME_DIR` target properties. They seem to be specific to MacOS and the docs say nothing about making them relative or default prefixes: https://cmake.org/cmake/help/latest/prop_tgt/INSTALL_NAME_DIR.html

Since I have no experience with MacOS I'm just leaving these with the absolute paths. If anything needs to be changed there it could also be addressed in a subsequent PR.

**Review by RainerKuemmerle — APPROVED** [review]

**Pull request merged**